ddtrace/opentelemetry/tracer_provider.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package opentelemetry
+
+import (
+	"sync"
+	"sync/atomic"
+
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var _ oteltrace.TracerProvider = (*TracerProvider)(nil)
+
+// TracerProvider provides implementation of OpenTelemetry TracerProvider interface.
+// TracerProvider provides Tracers that are used by instrumentation code to
+// trace computational workflows.
+type TracerProvider struct {
+	tracer *oteltracer
+	sync.Once
+	stopped atomic.Bool
+}
+
+const defaultName = "otel_datadog"
+
+// todo find a way to map Datadog options to Otel Options
+// - for tracer Start
+// - for span Start
+// - for span Finish
+
+// Tracer returns an instance of OpenTelemetry Tracer and initializes Datadog Tracer.
+func (p *TracerProvider) Tracer(name string, options ...oteltrace.TracerOption) oteltrace.Tracer {
```

Comment:
If we don't support [`WithInstrumentationVersion`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#WithInstrumentationVersion) or [`WithSchemaURL`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#WithSchemaURL) then let's call this out explicitly in the godoc for this method.
Let's also be clear in the docs about what we do with name.